lib/Db/Message.php

```diff
@@ -174,7 +174,9 @@ public function setInReplyTo(?string $inReplyTo): void {
 	}
 
 	public function setThreadRootId(?string $messageId): void {
-		$this->setMessageIdFieldIfNotEmpty('threadRootId', $messageId);
+		$threadRootId = (!empty($messageId)) ? '<' . rtrim(ltrim($messageId, '<'), '>') . '>' : $this->getMessageId();
+		$parsed = new Horde_Mail_Rfc822_Identification($threadRootId);
+		$this->threadRootId = ($parsed->ids[0] ?? null);
```

Comment:
```suggestion
		if (!empty($parsed->ids[0]) {
			$this->setter('threadRootId', $parsed->ids[0]);
		}
```

1) Does *not* set the thread root id to `null`
2) Works with the entity change detection. If you assign `$this->threadRootId` directly you bypass the call to `\OCP\AppFramework\Db\Entity::markFieldUpdated`.